### PR TITLE
Ete testcase update

### PR DESCRIPTION
**Pull request by zhulinJulia24**
1. fix workflow and split testcase step.
2. case can be executed in parallel.
3. add gpu status check, test case can be executed in free gpu.
4. add restful、pipeline and torch related testcases

### CODE REVIEW

File: .github/workflows/daily_ete_test.yml [hunk 1]
```diff
@@ -38,7 +35,7 @@ env:
 jobs:
   test_functions:
     runs-on: [self-hosted, linux-a100]
-    timeout-minutes: 4320 # 72hours
+    timeout-minutes: 180 # 3hours
     environment: 'prod'
```
Comment (RunningLeon) [inline .github/workflows/daily_ete_test.yml:39]: need to remove this line if it runs daily without human authorization

File: .github/workflows/daily_ete_test.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-name: test
+name: daily_ete_test
```
Comment (RunningLeon) [inline .github/workflows/daily_ete_test.yml:1]: may fix lint

Comment (RunningLeon) [inline .github/workflows/daily_ete_test.yml:1]: do you have a ran log on your forked repo?

Comment (zhulinJulia24) [reply .github/workflows/daily_ete_test.yml:1]: > ran

I‘m running on it, now it takes too long to run one round. I will attach it.

File: .github/workflows/daily_ete_test.yml [hunk 3]
```diff
@@ -63,7 +37,7 @@ jobs:
       - name: Install pytorch
         run: |
           python3 -m pip cache dir
-          python3 -m pip install torch==2.0.1 torchvision==0.15.2 --extra-index-url https://download.pytorch.org/whl/cu117
+          python3 -m pip install torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/cu118
```
Comment (RunningLeon) [inline .github/workflows/daily_ete_test.yml:40]: no need to install torchaudio

File: autotest/conftest.py [hunk 4]
```diff
@@ -13,40 +13,40 @@ def config(request):
...
-def restful_case_config(request):
-    case_path = os.path.join(request.config.rootdir,
-                             'restful_prompt_case.yaml')
+def common_case_config():
+    case_path = os.path.join(common_prompt_case_file)
     print(case_path)
     with open(case_path) as f:
         case_config = yaml.load(f.read(), Loader=yaml.SafeLoader)
-        del case_config['session_len_error']
+        # del case_config['session_len_error']
```
Comment (RunningLeon) [inline autotest/conftest.py:35]: could remove this line

File: autotest/README.md [hunk 5]
```diff
@@ -47,29 +46,35 @@ If you want to create a test environment, you need to prepare the above content
...
-|   [internlm_chat_20b](https://huggingface.co/internlm/internlm-chat-20b)   |        Yes         |   Yes   |       Yes        |        No        |       Yes       |
-|          [Qwen_7B_Chat](https://huggingface.co/Qwen/Qwen-7B-Chat)          |        Yes         |   Yes   |       Yes        |        No        |       Yes       |
-|         [Qwen_14B_Chat](https://huggingface.co/Qwen/Qwen-14B-Chat)         |        Yes         |   Yes   |       Yes        |        No        |       Yes       |
-| [Baichuan2_7B_Chat](https://huggingface.co/baichuan-inc/Baichuan2-7B-Chat) |        Yes         |   Yes   |       Yes        |        No        |       Yes       |
-|    [llama_2_7b_chat](https://huggingface.co/meta-llama/Llama-2-7b-chat)    |        Yes         |   Yes   |       Yes        |        No        |       Yes       |
+|                                   Models                                   | w4a16 quantization | w8a8 quantization | kvint quantization | convert | pipeline chat | pipeline chat - pytorch | restful_api chat | command chat - cli | command chat - hf | command chat - pytorch |
```
Comment (RunningLeon) [inline autotest/README.md:67]: `kvint quantization` -> `kvint8 quantization`

File: autotest/utils/pipeline_chat.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
+from lmdeploy import pipeline
+from lmdeploy.messages import (GenerationConfig, PytorchEngineConfig,
+                               TurbomindEngineConfig)
+
+
+class PipelineChat:
+
+    def __init__(self, hf_path: str, tp: int == 1):
```
Comment (RunningLeon) [inline autotest/utils/pipeline_chat.py:8]: `==` - > `=`

File: autotest/utils/pipeline_chat.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        self.pipe = pipe
+
+    def default_pipeline_chat(self, prompt):
+        gen_config = GenerationConfig(temperature=0.01)
+        return self.pipe([prompt], gen_config=gen_config)[0]
+
+
+class PipelinePytorchChat:
+
+    def __init__(self, hf_path: str, tp: int == 1):
```
Comment (RunningLeon) [inline autotest/utils/pipeline_chat.py:24]: `==` - > `=`

**Review by RunningLeon — APPROVED** [review]
LGTM

**Review by lvhan028 — APPROVED** [review]

**Pull request merged**